usvm-jvm/src/main/kotlin/org/usvm/machine/interpreter/statics/JcStaticFieldsRegion.kt

```diff
@@ -0,0 +1,148 @@
+import org.usvm.memory.UMemoryRegion
+import org.usvm.memory.UMemoryRegionId
+import org.usvm.memory.UReadOnlyMemory
+import org.usvm.memory.guardedWrite
+import org.usvm.sampleUValue
+import java.lang.reflect.Modifier
+
+data class JcStaticFieldLValue<Sort : USort>(
+    val field: JcField,
+    override val sort: Sort,
+) : ULValue<JcStaticFieldLValue<Sort>, Sort> {
+    override val memoryRegionId: UMemoryRegionId<JcStaticFieldLValue<Sort>, Sort> = JcStaticFieldRegionId(sort)
+
+    override val key: JcStaticFieldLValue<Sort>
+        get() = this
+}
+
+data class JcStaticFieldRegionId<Sort : USort>(
+    override val sort: Sort,
+) : UMemoryRegionId<JcStaticFieldLValue<Sort>, Sort> {
+    override fun emptyRegion(): UMemoryRegion<JcStaticFieldLValue<Sort>, Sort> = JcStaticFieldsMemoryRegion(sort)
+}
+
+internal class JcStaticFieldsMemoryRegion<Sort : USort>(
+    private val sort: Sort,
+    private var fieldValuesByClass: PersistentMap<JcClassOrInterface, PersistentMap<JcField, UExpr<Sort>>> = persistentHashMapOf(),
+) : UMemoryRegion<JcStaticFieldLValue<Sort>, Sort> {
+    val mutableStaticFields: List<JcField>
+        get() = fieldValuesByClass.values.flatMap { it.keys }.filter(fieldShouldBeSymbolic)
+
+    override fun read(key: JcStaticFieldLValue<Sort>): UExpr<Sort> {
+        val field = key.field
+        return fieldValuesByClass[field.enclosingClass]?.get(field)
+            ?: sort.jctx.mkStaticFieldReading(key.memoryRegionId as JcStaticFieldRegionId, field, sort)
+    }
+
+    override fun write(
+        key: JcStaticFieldLValue<Sort>,
+        value: UExpr<Sort>,
+        guard: UBoolExpr,
+    ): UMemoryRegion<JcStaticFieldLValue<Sort>, Sort> {
+        val field = key.field
+
+        val enclosingClass = field.enclosingClass
+        if (enclosingClass !in fieldValuesByClass) {
+            fieldValuesByClass = fieldValuesByClass.put(enclosingClass, persistentHashMapOf())
+        }
+
+        val newFieldValues = fieldValuesByClass
+            .getValue(enclosingClass)
+            .guardedWrite(key.field, value, guard) { key.sort.sampleUValue() }
+        val newFieldsByClass = fieldValuesByClass.put(enclosingClass, newFieldValues)
+
+        return JcStaticFieldsMemoryRegion(sort, newFieldsByClass)
+    }
+
+    fun mutatePrimitiveStaticFieldValuesToSymbolic(state: JcState, enclosingClass: JcClassOrInterface) {
+        val staticFields = fieldValuesByClass[enclosingClass] ?: return
+
+        val mutablePrimitiveStaticFieldsToSymbolicValues = staticFields
+            .entries
+            .filter { (field, _) -> fieldShouldBeSymbolic(field) }
+            .associate { (field, value) ->
+                val lvalue = JcStaticFieldLValue(field, value.sort)
+                val regionId = lvalue.memoryRegionId as JcStaticFieldRegionId
+                val symbol = state.ctx.mkStaticFieldReading(regionId, field, value.sort)
+
+                state.memory.write(lvalue, symbol, guard = state.ctx.trueExpr)
```

Comment:
Why do we need to write this symbol to the memory? Also, it is seems like we can simply remove all such fields from the region, instead of writing a bunch of symbols

Reply:
Fixed